item-api/src/main/java/com/example/auction/item/api/ItemService.java

```diff
@@ -21,67 +21,71 @@
+     * Start an auction for an item.
+     *
+     * @param id The id of the item to start the auction for.
+     * @return Done if the auction was started.
+     */
+    ServiceCall<NotUsed, Done> startAuction(UUID id);
 
-  /**
-   * Get an item with the given ID.
-   *
-   * @param id The ID of the item to get.
-   * @return The item.
-   */
-  ServiceCall<NotUsed, Item> getItem(UUID id);
+    /**
+     * Get an item with the given ID.
+     *
+     * @param id The ID of the item to get.
+     * @return The item.
+     */
+    ServiceCall<NotUsed, Item> getItem(UUID id);
 
-  /**
-   * Get a list of items for the given user.
-   *
-   * @param id The ID of the user.
-   * @param pageNo The page number.
-   * @param pageSize The number of items to return per page.
-   * @return The sequence of items.
-   */
-  ServiceCall<NotUsed, PSequence<Item>> getItemsForUser(UUID id, Optional<Integer> pageNo, Optional<Integer> pageSize);
+    /**
+     * Get a list of items for the given user.
+     *
+     * @param id       The ID of the user.
+     * @param status   The status of items to return.
+     * @param pageNo   The page number, starting from zero.
+     * @param pageSize The number of items to return per page.
+     * @return The sequence of items.
+     */
+    ServiceCall<NotUsed, PaginatedSequence<ItemSummary>> getItemsForUser(
+            UUID id, ItemStatus status, Optional<Integer> pageNo, Optional<Integer> pageSize);
 
-  /**
-   * The item events stream.
-   */
-  Topic<ItemEvent> itemEvents();
+    /**
+     * The item events stream.
+     */
+    Topic<ItemEvent> itemEvents();
 
-  @Override
-  default Descriptor descriptor() {
-    return named("item").withCalls(
-            pathCall("/api/item", this::createItem),
-            restCall(Method.POST, "/api/item/:id/start", this::startAuction),
-            pathCall("/api/item/:id", this::getItem),
-            pathCall("/api/item?userId&pageNo&pageSize", this::getItemsForUser)
-    ).publishing(
-            topic("item-ItemEvent", this::itemEvents)
-    ).withPathParamSerializer(UUID.class, PathParamSerializers.required("UUID", UUID::fromString, UUID::toString))
-            .withHeaderFilter(SecurityHeaderFilter.INSTANCE);
-  }
+    @Override
+    default Descriptor descriptor() {
+        return named("item").withCalls(
+                pathCall("/api/item", this::createItem),
+                restCall(Method.POST, "/api/item/:id/start", this::startAuction),
+                pathCall("/api/item/:id", this::getItem),
+                pathCall("/api/item?userId&status&pageNo&pageSize", this::getItemsForUser)
+        ).publishing(
+                topic("item-ItemEvent", this::itemEvents)
+        ).withPathParamSerializer(
+                UUID.class, PathParamSerializers.required("UUID", UUID::fromString, UUID::toString)
+        ).withPathParamSerializer(
+                ItemStatus.class, PathParamSerializers.required("ItemStatus", ItemStatus::valueOf, ItemStatus::toString)
```

Comment:
Yeah - though currently lookup of serializers is done via class match, not via class heirarchy.  I think though it's all done ahead of time, so there's no harm in doing a dumb iteration through all classes and doing a `isAssignableFrom` or whatever. We should just do the exact class match first.